### ISSUE TITLE
Bug when recovering from seed

### ISSUE THREAD

**Issue by shyrwall** [monero-project/monero#6581]
When recovering a wallet from a seed it isn't clear if all the 25 words should be on the same row or if you can enter the words one by one. And if you do enter the words one by one then only 24 are accepted + an offset. See below,

Generating new wallet...
1 Specify Electrum seed:
2 Electrum seed continued:
3 Electrum seed continued:
4 Electrum seed continued:
5 Electrum seed continued:
6 Electrum seed continued:
7 Electrum seed continued:
8 Electrum seed continued:
9 Electrum seed continued:
10 Electrum seed continued:
11 Electrum seed continued:
12 Electrum seed continued:
13 Electrum seed continued:
14 Electrum seed continued:
15 Electrum seed continued:
16 Electrum seed continued:
17 Electrum seed continued:
18 Electrum seed continued:
19 Electrum seed continued:
20 Electrum seed continued:
21 Electrum seed continued:
22 Electrum seed continued:
23 Electrum seed continued:
24 Electrum seed continued:
25 Enter seed offset passphrase, empty if none:

--

I suggest either allowing only the seed to be entered on 1 line (and maybe show with *) or just fix the bug and also allow multi line entering.

**Comment by moneromooo-monero** [comment]
You can do either one line for one word per line, either should be accepted.
Some seeds are 24 words, that's why it doesn't require a 25th if there is none. I suppose it could require an empty line in that case, but it seems complicating for no good reason.

**Comment by selsta** [comment]
Did monero ever have 24 word seeds? AFAIK that’s only because dropping the offset is also valid. I think the confusion was that @shyrwall thought the seed offset is the 25th word.

**Comment by moneromooo-monero** [comment]
Well, yes, before the checksum got added :)

**Comment by shyrwall** [comment]
> You can do either one line for one word per line, either should be accepted.
> Some seeds are 24 words, that's why it doesn't require a 25th if there is none. I suppose it could require an empty line in that case, but it seems complicating for no good reason.

The problem is the 25 word seed doesn't work line by line :) Since it's expecting 24 words and then the 25th is the passphrase.

**Comment by shyrwall** [comment]
Or do you mean that if doing line by line the 25th word should not be entered at all? Because its a checksum? So the offset should just be empty.

If so I got it..

Unclear but I get it :D

**Comment by iDunk5400** [comment]
It looks like `25` and `Enter seed offset passphrase, empty if none:` should be on separate lines.

**Comment by moneromooo-monero** [comment]
I kinda skipped over the numbers mentally. I don't know where they're coming from, unless the output was manually altered to add them.

**Comment by iDunk5400** [comment]
Oh right, I didn't consider that.

**Comment by moneromooo-monero** [comment]
For the record, since it's no obvious, the reason why it accepts several lines is because the seed is displayed on three lines, so it gets copy/pasted to a file for backup, so I expect it usually gets copy/pasted back as three lines.

**Comment by shyrwall** [comment]
> I kinda skipped over the numbers mentally. I don't know where they're coming from, unless the output was manually altered to add them.

Yes i just added the numbers manually so you could see the number of rows.

**Comment by shyrwall** [comment]
If continuing the multiline input possibility this seems like a fix. Tested but not familiar enough with the code to know other impacts

--- simplewallet.cpp        2020-05-22 05:46:08.255804804 +0200
+++ simplewallet.cpp    2020-05-22 05:46:21.383585057 +0200
@@ -3904,7 +3904,7 @@
   std::vector<epee::wipeable_string> seed;

   words.split(seed);
-  return seed.size() < 24;
+  return seed.size() < 25;
 }
 //----------------------------------------------------------------------------------------------------
 static bool datestr_to_int(const std::string &heightstr, uint16_t &year, uint8_t &month, uint8_t &day)

**Comment by moneromooo-monero** [comment]
Would that not prevent 24 word seed from being used ?

**Comment by moneromooo-monero** [comment]
I checked, it would. But maybe one could add a check for an empty line, to say "I'm really done".
On the other hand, that 25th word is not needed in the first place, so I'm not sure it's worth the effort.

**Comment by shyrwall** [comment]
Ah.  Didnt even know there used to be 24 word seeds. 

I'm just thinking of it from a user aspect and most would have no idea the last one is a checksum and could be excluded. I sure didnt know and i would think i'm above the average joe :D

Just a suggestion to alleviate support request and/or thinking they lost the funds.

But up to you guys if its worth any change :)

**Comment by moneromooo-monero** [comment]
OK. I think the "empty line stops" should work for both cases. You're the first ever to report that fwiw :)
